### PR TITLE
Feat 807 validator add remove signer

### LINKED ISSUE #807: Validator add and remove signer
Provide a UI for validators to add themselves as a signer of the multisig contract. Could consist of a simple one field form page where you can input your node ID and then execute an Ethereum transaction against the multisig control contract.

Contract function: https://github.com/vegaprotocol/MultisigControl/blob/develop/contracts/MultisigControl.sol#L50-L61

Currently blocked by: https://github.com/vegaprotocol/vega/issues/5678 however the protos can be found here: https://github.com/vegaprotocol/protos/blob/develop/sources/data-node/api/v2/trading_data.proto#L493-L573

### PR DESCRIPTION
**Pull request by sam-keen**
# Related issues 🔗

Closes #807

# Description ℹ️

A small simple app for adding and removing signers from the multisig contract.

# Demo 📺

Screenshot 2022-09-29 at 14 32 50

# Technical 👨‍🔧

Note: this cannot be tested yet as we don't have validators coming and going from a network. Tests are very thin on the ground for this first pass.

### CODE REVIEW

File: apps/multisig-signer/src/main.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import App from './app/app';
+import { StrictMode } from 'react';
+
+const rootElement = document.getElementById('root');
+const root = rootElement && createRoot(rootElement);
+
+root?.render(
+  <StrictMode>
+    <BrowserRouter>
```
Comment (dexturr) [inline apps/multisig-signer/src/main.tsx:13]: chore (blocking): let's remove router here as it is not needed

File: apps/multisig-signer-e2e/.env [hunk 2]
```diff
@@ -0,0 +1,4 @@
+NX_VEGA_URL=https://api.n01.stagnet3.vega.xyz/graphql
```
Comment (dexturr) [inline apps/multisig-signer-e2e/.env:1]: chore (blocking): let's kill this whole directory as it is unused

File: apps/multisig-signer/src/app/app.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import App from './app';
```
Comment (dexturr) [inline apps/multisig-signer/src/app/app.spec.tsx:1]: chore (blocking): dead code, delete

File: apps/multisig-signer/src/app/lib/apollo-client.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  });
+
+  const httpLink = new HttpLink({
+    uri: urlHTTP.href,
+    credentials: 'same-origin',
+  });
+
+  const errorLink = onError(({ graphQLErrors, networkError }) => {
+    console.log(graphQLErrors);
+    console.log(networkError);
```
Comment (dexturr) [inline apps/multisig-signer/src/app/lib/apollo-client.tsx:44]: chore (blocking): add sentry.capture exception here

**Review by dexturr — APPROVED** [review]

File: apps/multisig-signer/src/app/config/flags.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { ENV } from './env';
+
+export default {
+  ...ENV.flags,
```
Comment (mattrussell36) [inline apps/multisig-signer/src/app/config/flags.ts:4]: chore (non blocking): Unnecessary spread

File: apps/multisig-signer/src/app/config/index.tsx [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import { ENV } from './env';
+
+export const DATA_SOURCES = {
+  ...ENV.dataSources,
```
Comment (mattrussell36) [inline apps/multisig-signer/src/app/config/index.tsx:4]: chore (non blocking): Unnecessary spread

File: apps/multisig-signer/src/app/config/contracts/contracts-provider.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+export const ContractsProvider = ({ children }: { children: JSX.Element }) => {
+  const { config } = useEthereumConfig();
+  const { VEGA_ENV, ETHEREUM_PROVIDER_URL } = useEnvironment();
+  const [contracts, setContracts] =
+    React.useState<ContractsContextShape | null>(null);
+
+  // Create instances of contract classes. If we have an account use a signer for the
+  // contracts so that we can sign transactions, otherwise use the provider for just
+  // reading data
+  React.useEffect(() => {
```
Comment (mattrussell36) [inline apps/multisig-signer/src/app/config/contracts/contracts-provider.tsx:22]: chore (non blocking): for consistentcy import and use `useEffect` directly, same for other react imports in other files.

File: apps/multisig-signer/src/app/components/web3-connector/web3-connector.tsx [hunk 8]
```diff
@@ -0,0 +1,94 @@
+import { useEnvironment } from '@vegaprotocol/environment';
+import { useEthereumConfig } from '@vegaprotocol/web3';
+import { Button, Splash, AsyncRenderer } from '@vegaprotocol/ui-toolkit';
+import { Web3ConnectDialog } from '@vegaprotocol/web3';
```
Comment (mattrussell36) [inline apps/multisig-signer/src/app/components/web3-connector/web3-connector.tsx:4]: chore (blocking): Combine web3 imports

File: apps/multisig-signer/src/app/components/web3-connector/web3-connector.tsx [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+import { useEffect, useMemo } from 'react';
+import { createConnectors } from '../../lib/web3-connectors';
+
+interface Web3ConnectorProps {
+  children: ReactElement;
+  dialogOpen: boolean;
+  setDialogOpen: (open: boolean) => void;
+}
+
+export function Web3Connector({
```
Comment (mattrussell36) [inline apps/multisig-signer/src/app/components/web3-connector/web3-connector.tsx:16]: question (blocking): why not use web3-container from web3 lib? This file looks _almost_ the same. If the one in the lib isn't suitable, why? and can we update it to be better fit for purpose

File: apps/multisig-signer/src/app/components/remove-signer-form/remove-signer-form.tsx [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+  >(REMOVE_SIGNER_QUERY);
+  const { perform, Dialog } = useEthereumTransaction<
+    MultisigControl,
+    'remove_signer'
+  >(multisig, 'remove_signer');
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    setBundleNotFound(false);
+    try {
+      if (address === '') {
```
Comment (mattrussell36) [inline apps/multisig-signer/src/app/components/remove-signer-form/remove-signer-form.tsx:52]: suggestion (non blocking): could just use react hook form here to get some easier client side validation. Deffo follow up though

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/multisig-signer/src/app/components/web3-connector/web3-connector.tsx (hunk 8 above)
Comment (mattrussell36) [reply apps/multisig-signer/src/app/components/web3-connector/web3-connector.tsx:4]: After discussion, we will switch this out later: https://github.com/vegaprotocol/frontend-monorepo/issues/1550

File: apps/multisig-signer/src/app/components/remove-signer-form/remove-signer-form.tsx (hunk 10 above)
Comment (sam-keen) [reply apps/multisig-signer/src/app/components/remove-signer-form/remove-signer-form.tsx:52]: https://github.com/vegaprotocol/frontend-monorepo/issues/1559

File: apps/multisig-signer/src/app/components/web3-connector/web3-connector.tsx (hunk 8 above)
Comment (mattrussell36) [reply apps/multisig-signer/src/app/components/web3-connector/web3-connector.tsx:4]: Oops replied to the wrong comment

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**